src/messages/Message.ts

```diff
@@ -131,6 +151,8 @@ export default class Message {
       ? bytesToByteArrayWithSize(this._encryptedData, 'int32')
       : concatBytes(stringToByteArrayWithSize(this.mediaType), bytesToByteArrayWithSize(this.data, 'int32'));
 
+    const messageInfoBytes = stringToByteArrayWithSize(JSON.stringify(this.messageInfo || {}));
```

Comment:
Don't JSON encode this. The purpose of `toBinary` is to keep the data binary instead of using JSON / base64 encoding.

Use `stringToByteArrayWithSize` for the title and description.
The thumbnail should already be a `Binary`. Prepend the binary with the size.